examples/theme-ui/app/root.tsx

```diff
@@ -40,7 +40,7 @@ const Document = withEmotionCache(
       });
       // reset cache to reapply global styles
       clientStyleData?.reset();
-    }, [clientStyleData, emotionCache.sheet]);
+    }, []);
```

Comment:
This causes an ESLint warning
https://github.com/remix-run/remix/runs/7713555214?check_suite_focus=true#step:6:11

```
Warning:   43:8  warning  React Hook useEffect has missing dependencies: 'clientStyleData' and 'emotionCache.sheet'. Either include them or remove the dependency array  react-hooks/exhaustive-deps
```

Can we make sure to not have an `eact-hooks/exhaustive-deps` warning + not disable it, but make the example work please?

Reply:
Hmm. I had a go, I've been unable to find a solution that doesn't involve infinite looping. I'm going to keep trying, it might need a bit of refactoring to achieve.
In the meantime, do you think it's worth raising the loop issue separately? So we can fix the styling without it also depending on this tricky change.

Reply:
I'd love to either have this fixed _or_ set the dependency array correct again.